worker-message-prioritization-distribution/src/test/java/com/github/workerframework/workermessageprioritization/redistribution/consumption/StagingQueueWeightSettingsTest.java

```diff
@@ -0,0 +1,213 @@
+        envVariables.put("CAF_ADJUST_QUEUE_WEIGHT_5", "repository\\-initialization\\-workflow$,0.5");
+
+        try(MockedStatic<EnvVariableCollector> envVariableCollectorMock = Mockito.mockStatic(EnvVariableCollector.class)) {
+
+            envVariableCollectorMock.when(EnvVariableCollector::getEnvVariables).thenReturn(envVariables);
+            envVariableCollectorMock.when(() -> EnvVariableCollector.getQueueWeightEnvVariables(anyMap())).thenReturn(envVariables);
+
+            final StagingQueueWeightSettingsProvider stagingQueueWeightSettingsProvider =
+                    new StagingQueueWeightSettingsProvider();
+
+            final List<String> stagingQueueNames =
+                    distributorWorkItem.getStagingQueues().stream().map(Queue::getName).collect(toList());
+
+            final Map<String, Double> stagingQueueWeightMap =
+                    stagingQueueWeightSettingsProvider.getStagingQueueWeights(stagingQueueNames);
+
+            assertEquals("Weight of queue should be set by environment variable.",
+                    10, stagingQueueWeightMap.get("bulk-indexer-in»/clynch/enrichment-workflow"), 0.0);
+            assertEquals("Weight of queue should be set by environment variable which matches the longest length of string.",
+                    10, stagingQueueWeightMap.get("bulk-indexer-in»/rory3/enrichment-workflow"), 0.0);
+            assertEquals("Weight of queue should be set by environment variable.",
+                    0, stagingQueueWeightMap.get("dataprocessing-classification-in»/clynch/update-entities-workflow"), 0.0);
+            assertEquals("Weight of queue should be set by environment variable.",
+                    3, stagingQueueWeightMap.get("dataprocessing-classification-in»/a77777/update-entities-workflow"), 0.0);
+            assertEquals("No weight set to match this string therefore should default to 1.",
+                    1, stagingQueueWeightMap.get("dataprocessing-classification-in»/rory3/update-entities-workflow"), 0.0);
+            assertEquals("Two strings matched of same length with different weights should set to larger weight.",
+                    3, stagingQueueWeightMap.get("bulk-indexer-in»/clynch/a77777"), 0.0);
+            assertEquals("Weight of queue should be set by environment variable.",
+                    7, stagingQueueWeightMap.get("dataprocessing-langdetect-in»/mahesh/ingestion-workflow"), 0.0);
+            assertEquals("Weights can be set below 1 to reduce the processing of the queue.",
+                    0.5, stagingQueueWeightMap.get("bulk-indexer-in»/jmcc02/repository-initialization-workflow"), 0.0);
+        }
+    }
+
+    @Test
+    public void passIncorrectEnvVariableFormatWithSpaceTest() {
+
+        final Queue targetQueue = getQueue("tq", 1000);
+
+        final Queue q1 = getQueue("bulk-indexer-in»/clynch/enrichment-workflow", 1000);
+
+        final Set<Queue> stagingQueues = new HashSet<>(Arrays.asList(q1));
```

Comment:
Arrays.asList > Collections.singletonList (when there's only on thing in the list)